### DISCUSSION TITLE
How to create private folders in public wiki

### DISCUSSION THREAD

**Discussion by tricyphona** [requarks/wiki#6579] [Help / Questions] [answered]
Wiki.js version: 2.5.298
OS: Docker container
Node.js: 16.20.0
Database: MariaDB 10.6.12

The goal is to have private folders inside public folders.
Say we have a structure of:
/publicFile/publicFile1
/publicFile/publicFile1/privateFile1
/publicFile/publicFile1/privateFile1/privateFile1.1
/publicFile/publicFile1/privateFile2
/publicFile/publicFile2/
/publicFile/publicFile2/privateFile1
[...]

Members of a group can view the privatefile1, general members can view everything until /publicFile/PublicFile[1,2,3,4,5] etc. But not the content of those directories. 
Current permissions for group "Members" are: 
Allow read for starting with /
Deny read path starts with /publicFile/publicFile1/

Added a tag publicFile1 for the private pages under publicFile1, 
Permissions for group "readPublicFolder1"
Allow read tag matches publicFile1

All users are part of "members" and some also part of "readPublicFolder1". 
Current situation: both groups can't read /publicFile/publicFile1/privateFile1.
Desired situation: All members can read public files. Users need a specific group to get read access to read the private files within /publicFile/publicFile1/* (And another group membership for /publicFile/publicFile2/*)
Is such a usecase supported and if so, how would I get my desired situation?

**Comment by P4mes4n** [comment, marked as answer]
Take a look at this (taken from [official docs](https://docs.requarks.io/groups)):
```
When 2 rules have the same specificity, the priority is given from lowest to highest as follows:

Path Starts With... (lowest)
Path Ends With...
Path Matches Regex...
Path Is Exactly... (highest)

When 2 rules have the same path specificity AND the same match type, Deny will always override an Allow rule.
```

Tags aren't listed in the priority. Probably because permissions via tags were added later on.
Try to change the ALLOW rule on your `readPublicFolder1` to a higher priority matching.

**Reply by tricyphona** [reply]
Thank you! 
I had read over it that tags were not included in the order of rules.
Current setup is 
group Members:
Allow read startswith /
Deny Read startswith /publicfile/publicfile1/

group folder:
Allow Read regex /publicfile/publicfile1/*

And gives desired result.
